iox_objectstore_garbage_collect/src/lib.rs

```diff
@@ -0,0 +1,232 @@
+//! Tool to clean up old object store files that don't appear in the catalog.
+
+#![deny(
+    rustdoc::broken_intra_doc_links,
+    rust_2018_idioms,
+    missing_debug_implementations,
+    unreachable_pub
+)]
+#![warn(
+    missing_docs,
+    clippy::todo,
+    clippy::dbg_macro,
+    clippy::clone_on_ref_ptr,
+    clippy::future_not_send
+)]
+#![allow(clippy::missing_docs_in_private_items)]
+
+use chrono::{DateTime, Utc};
+use chrono_english::{parse_date_string, Dialect};
+use clap::Parser;
+use clap_blocks::{
+    catalog_dsn::CatalogDsnConfig,
+    object_store::{make_object_store, ObjectStoreConfig},
+};
+use iox_catalog::interface::Catalog;
+use object_store::DynObjectStore;
+use observability_deps::tracing::*;
+use snafu::prelude::*;
+use std::sync::Arc;
+use tokio::sync::mpsc;
+use trogging::cli::LoggingConfig;
+
+/// Logic for checking if a file in object storage should be deleted or not.
+pub mod checker;
+/// Logic for deleting a file from object storage.
+pub mod deleter;
+/// Logic for listing all files in object storage.
+pub mod lister;
+
+const BATCH_SIZE: usize = 1000;
+
+/// Run the tasks that clean up old object store files that don't appear in the catalog.
+pub async fn inner_main(config: Config) -> Result<()> {
```

Comment:
```suggestion
pub async fn main(config: Config) -> Result<()> {
```

"inner" suggests this is private which is no longer the case.